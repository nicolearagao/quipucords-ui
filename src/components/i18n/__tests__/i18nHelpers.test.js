import React from 'react';
import { i18nHelpers, EMPTY_CONTEXT, translate } from '../i18nHelpers';

describe('I18nHelpers', () => {
  it('should have specific functions', () => {
    expect(i18nHelpers).toMatchSnapshot('i18nHelpers');
  });

  it('should attempt to perform translate with a node', async () => {
    const ExampleComponent = () => <div>{translate('lorem.ipsum', { hello: 'world' }, [<span id="test" />])}</div>;
    ExampleComponent.propTypes = {};
    ExampleComponent.defaultProps = {};

    const component = renderComponent(<ExampleComponent />);
    expect(component).toMatchSnapshot('translated node');
  });

  it('should attempt to perform a string replace', () => {
    const emptyContext = translate('lorem.ipsum', { context: EMPTY_CONTEXT });
    const emptyPartialContext = translate('lorem.ipsum', { context: ['hello', EMPTY_CONTEXT] });
    const localeKey = translate('lorem.ipsum');
    const placeholder = translate('lorem.ipsum', 'hello world');
    const multiContext = translate('lorem.ipsum', { context: ['hello', 'world'] });
    const multiContextWithEmptyValue = translate('lorem.ipsum', { context: ['hello', undefined, null, '', 'world'] });
    const multiKey = translate(['lorem.ipsum', undefined, null, '', 'lorem.fallback']);

    expect({
      emptyContext,
      emptyPartialContext,
      localeKey,
      placeholder,
      multiContext,
      multiContextWithEmptyValue,
      multiKey
    }).toMatchSnapshot('translate');
  });
});
