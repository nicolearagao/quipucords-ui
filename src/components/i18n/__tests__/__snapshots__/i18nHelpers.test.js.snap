// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18nHelpers should attempt to perform a string replace: translate 1`] = `
{
  "emptyContext": "t(lorem.ipsum, {"context":" "})",
  "emptyPartialContext": "t(lorem.ipsum_hello, {"context":" "})",
  "localeKey": "t(lorem.ipsum)",
  "multiContext": "t(lorem.ipsum_hello, {"context":"world"})",
  "multiContextWithEmptyValue": "t(lorem.ipsum_hello, {"context":"world"})",
  "multiKey": "t([lorem.ipsum,lorem.fallback])",
  "placeholder": "t(lorem.ipsum, hello world)",
}
`;

exports[`I18nHelpers should attempt to perform translate with a node: translated node 1`] = `
<examplecomponent
  props="{
  "children": []
}"
>
  <div>
    t(lorem.ipsum, {"hello":"world"}, [object Object])
  </div>
</examplecomponent>
`;

exports[`I18nHelpers should have specific functions: i18nHelpers 1`] = `
{
  "EMPTY_CONTEXT": "LOCALE_EMPTY_CONTEXT",
  "translate": [Function],
}
`;
